### PR TITLE
Migrate to using Johnny-Five's board mounting!

### PR DESCRIPTION
**Pull request by reconbot**
Change the pin constructor to no longer require the board or five objects.
- We now have a new option of an optional board
- Bonus! Protect against accidently non construction
- Add some docs around options

### CODE REVIEW

**Comment by noopkat** [comment]
fab fab fab, lemme test this out and let you know! :sparkles:

**Comment by rwaldron** [comment]
You shouldn't use pins named like that at all. Use a property `pins: { reset: ..., slave: ...}}`

File: oled.js [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
+    return new Oled(opts);
+  }
+
+  five.Board.Component.call(
+    this, opts = five.Board.Options(opts)
+  );
+
+  var pins = opt.pins || {};
+  this.SLAVEPIN = pins.slave || 12;
+  this.RESETPIN = pins.reset || 4;
```
Comment (rwaldron) [inline oled.js]: `Board.Component` will have already made `this.pins.slave` etc

**Comment by noopkat** [comment]
@reconbot @rwaldron thanks for these code changes for the better, you're both saints :wine_glass:  I'm going to test this out when I get a chance. I have a permanent OLED rig now which is nice, even if it's still a bit to go with automating it :wink:

**Pull request closed without merging**